firmware_configurations/klipper/IamPerrine/SKR_Fade_Config.txt

```diff
@@ -0,0 +1,751 @@
+
+######################################################################
+# Filament Change
+######################################################################
+
+# M600: Filament Change. This macro will pause the printer, move the
+# tool to the change position, and retract the filament 50mm. Adjust
+# the retraction settings for your own extruder. After filament has
+# been changed, the print can be resumed from its previous position
+# with the "RESUME" gcode.
+
+######################################################################
+# Filament sensors
+######################################################################
+
+# Filament Switch Sensor.  Support for filament insert and runout detection
+# using a switch sensor, such as an endstop switch.
+[filament_switch_sensor my_sensor]
+pause_on_runout: True
+#   When set to True, a PAUSE will execute immediately after a runout
+#   is detected. Note that if pause_on_runout is False and the
+#   runout_gcode is omitted then runout detection is disabled. Default
+#   is True.
+runout_gcode:
+	M117 Runout Detected
+	M600
+	M400
+#   A list of G-Code commands to execute after a filament runout is
+#   detected. See docs/Command_Templates.md for G-Code format. If
+#   pause_on_runout is set to True this G-Code will run after the
+#   PAUSE is complete. The default is not to run any G-Code commands.
+#insert_gcode:
+#	M117 Insert Detected
+#   A list of G-Code commands to execute after a filament insert is
+#   detected. See docs/Command_Templates.md for G-Code format. The
+#   default is not to run any G-Code commands, which disables insert
+#   detection.
+event_delay: 3.0
+#   The minimum amount of time in seconds to delay between events.
+#   Events triggered during this time period will be silently
+#   ignored. The default is 3 seconds.
+pause_delay: 0.5
+#   The amount of time to delay, in seconds, between the pause command
+#   dispatch and execution of the runout_gcode.  It may be useful to
+#   increase this delay if Octoprint exhibits strange pause behavior.
+#   Default is 0.5 seconds.
+switch_pin: !P1.25
+#   The pin on which the switch is connected. This parameter must be
+#   provided.
+
+# Change XYZ to desired position hotend parking location
+[gcode_macro M600]
+default_parameter_X: 175
+default_parameter_Y: 1
+default_parameter_Z: 50
+gcode:
+    M117 Filament Change
+	SAVE_GCODE_STATE NAME=M600_state
+    PAUSE
+    G91
+    G1 E-5 F4000
+	G1 Z{Z}
+	G1 E-700 F5000
+    G90
+    G1 X{X} Y{Y} F3000
+    G91
+    RESTORE_GCODE_STATE NAME=M600_state
+
+[gcode_macro SWAP_RESUME]
+gcode:
+ 	M117 Printing...
+	LOAD_FILAMENT
+	RESUME
+	
+# Sensor Types
+#   "EPCOS 100K B57560G104F"
+#   "ATC Semitec 104GT-2"
+#   "NTC 100K beta 3950"
+#   "Honeywell 100K 135-104LAG-J01"
+#   "NTC 100K MGB18-104F39050L32" (Keenovo Heater Pad)
+#   "AD595"
+#   "PT100 INA826"
+
+#*# <---------------------- SAVE_CONFIG ---------------------->
```

Comment:
This block will likely not conform to any other printer but your own. If someone chooses to use this config, they might miss that this block is present and get unexpected behavior. I suggest to remove them.